packages/bundler/src/Bundler.ts

```diff
@@ -105,27 +133,83 @@ export class Bundler implements IBundler {
       userOpHash: sendUserOperationResponse.result,
       wait: (confirmations?: number): Promise<UserOpReceipt> => {
         const provider = new JsonRpcProvider(RPC_PROVIDER_URLS[chainId]);
+        // Note: maxDuration can be defined per chainId
+        const maxDuration = this.UserOpReceiptMaxDurationIntervals[chainId] || 30000; // default 30 seconds
+        let totalDuration = 0;
+
         return new Promise<UserOpReceipt>((resolve, reject) => {
-          const intervalId = setInterval(async () => {
-            try {
-              const userOpResponse = await this.getUserOpReceipt(sendUserOperationResponse.result);
-              if (userOpResponse && userOpResponse.receipt && userOpResponse.receipt.blockNumber) {
-                if (confirmations) {
-                  const latestBlock = await provider.getBlockNumber();
-                  const confirmedBlocks = latestBlock - userOpResponse.receipt.blockNumber;
-                  if (confirmations >= confirmedBlocks) {
+          if (this.UserOpReceiptIntervals && chainId in this.UserOpReceiptIntervals) {
```

Comment:
Just like we defined a default value of 30seconds for maxDuration, we can also define the default value for UserOpReceiptIntervals as well and remove this If condition.. to not have hard dependency on sdk to support new network.